client/lib/media/actions.js

```diff
@@ -213,23 +219,37 @@ MediaActions.edit = function( siteId, item ) {
 };
 
 MediaActions.update = function( siteId, item ) {
-	var newItem;
-
 	if ( Array.isArray( item ) ) {
 		item.forEach( MediaActions.update.bind( null, siteId ) );
 		return;
 	}
 
-	newItem = assign( {}, MediaStore.get( siteId, item.ID ), item );
+	const mediaId = item.ID;
+	const newItem = assign( {}, MediaStore.get( siteId, item.ID ), item );
 
-	Dispatcher.handleViewAction( {
+	// let's updated data immediately
+	const updateAction = {
 		type: 'RECEIVE_MEDIA_ITEM',
-		siteId: siteId,
+		siteId,
 		data: newItem
-	} );
+	};
+
+	if ( item.media ) {
+		// also print edited image as soon as possible
+		updateAction.data = Object.assign(
+			{},
+			newItem,
+			MediaActions.createTransientMedia( mediaId, item.media )
+		);
+	}
+
+	debug( 'Updating media for %o by ID %o to %o', siteId, mediaId, updateAction );
+	Dispatcher.handleViewAction( updateAction );
 
-	debug( 'Updating media for %d by ID %d to %o', siteId, item.ID, item );
-	wpcom.site( siteId ).media( item.ID ).update( item, function( error, data ) {
+	wpcom
+	.site( siteId )
```

Comment:
Hmm interesting, we usually indent chains in our code examples, but doing so here will throw an eslint warning.